ARCollectionViewMasonryLayout.m

```diff
@@ -156,8 +163,15 @@ - (void)setupLayoutWithStaticDimension:(CGFloat)staticDimension andVariableDimen
     if (headerDimension != NSNotFound) {
         [self setupHeaderAtIndexPath:indexPathZero];
         leadingInset += headerDimension;
-    } else {
-        self.headerAttributes = nil;
+        [invalidationContext invalidateSupplementaryElementsOfKind:UICollectionElementKindSectionHeader atIndexPaths:@[indexPathZero]];
```

Comment:
instead of keeping these attributes in context, we now declare to the layout that we need to get the stuff for the header/footer